examples/security/secrets/README.md

```diff
@@ -0,0 +1,62 @@
+# Secrets
+
+Support for application secrets was added to wasmCloud 1.1.0 after the community accepted [RFC #2190](https://github.com/wasmCloud/wasmCloud/issues/2190).
+
+This folder has an example of an application that has a capability provider and component that refer to a secret value. The secret itself is encrypted and stored in a [NATS KV secrets backend](../../../crates/secrets-nats-kv/) instance, and wasmCloud fetches this secret at runtime based on the signed identity of the component/provider.
+
+**This is a modified version of the [http-keyvalue-counter](../../rust/components/http-keyvalue-counter/) application that authenticates with a Redis database that requires a password, and serves an HTTP API that requires an authentication password header.**
+
+This example uses a locally built version of the NATS KV secrets backend and wasmCloud from the `main` branch of this repository, as wasmCloud 1.1 is not yet released. Additionally, support for secrets in the [wadm](https://github.com/wasmCloud/wadm) application manifest was not released at the time of writing this example, so the application and setup is done imperatively. We'll update this example to also have an application manifest for easy setup once available.
+
+## Prerequisites
+
+- [Rust toolchain](https://www.rust-lang.org/tools/install)
+- [wash](https://wasmcloud.com/docs/installation)
+- [NATS CLI](https://github.com/nats-io/natscli)
+- [redis-server](https://redis.io/docs/latest/operate/oss_and_stack/install/install-redis/)
+- [nats-server](https://docs.nats.io/running-a-nats-service/introduction)
+- [jq](https://jqlang.github.io/jq/download/)
+
+## Running this example
+
+Build wasmCloud, the NATS KV secrets backend, the keyvalue counter auth component, and the keyvalue redis password provider:
+
+```bash
+./build.sh
```

Comment:
What do you think about doing this without `bash`? If we were going to do bash scripts I'd prefer `Makefile`s to start with... and if we were going to do `Makefile`s, I'd prefer `Justfile`s...

Since no extra tooling would be best, this might be a great place to actually introduce https://github.com/matklad/cargo-xtask

Reply:
It's hard since both bash and make have difficulties running on windows, so there's not a great option there. I would personally love to condense this down to either a cargo example or a much smaller set of runnable commands in the README, but for now this example is just too complicated to setup (lack of wadm support for secrets, running binaries from main, etc.)

I think this example has quite a long way to go before it is polished but I would ideally like to split it out into different README steps once it's complete. I was thinking the current scripts are just a little too long to do that now though

Reply:
Yeah I can see that -- let's punt on this